File: pyg-manager-interface/src/main/java/com/pyg/managerinterface/SpecificationService.java
package com.pyg.managerinterface;



import com.pyg.bean.PageResult;
import com.pyg.bean.ResultMessage;
import com.pyg.pojo.Specification;
import com.pyg.pojo.TbSpecification;

import java.util.List;
import java.util.Map;

/**
 * @author Technicolor
 * @version v1.0
 * @date 2019/3/7 21:39
 * @description TODO
 **/


public interface SpecificationService {

    /**
     * 查询所有品牌
     * @return
     */
    List<TbSpecification> findAll();

    /**
     * 分页查询
     * @param pageNo
     * @param pageSize
     * @return
     */
    PageResult findPage(Integer pageNo, Integer pageSize);

    /**
     * 新增品牌
     * @param specification
     * @return
     */
    void add(Specification specification);

    /**
     * 修改数据回现
     * @param id
     * @return
     */
    Specification findOne(Long id);

    /**
     * 根据id修改品牌
     * @param specification
     * @return
     */
    void update(Specification specification);

    /**
     * 根据id删除
     * @param ids
     * @return
     */
    void dele(Long[] ids);

    /**
     * 模糊分页查询
     * @param pageNo
     * @param pageSize
     * @param specification
     * @return
     */

    PageResult queryPage(Integer pageNo, Integer pageSize, TbSpecification specification);


    /**
     * 此方法是添加模板时要求的
     * @return
     */
    List<Map> findSpecList();

}

File: pyg-manager-web/src/main/webapp/js/service/specificationService.js
// 业务逻辑层和后台数据交互
app.service("specificationService",function ($http) {



    /**
     * 此方法是添加模板时要求的，数据格式为[{"id":1,"text":"联想"},{"id":2,"text":"华为"}]
     */
    this.findSpecList=function () {
        return $http.get("../specification/findSpecList");
    };

    this.findAll = function () {         //创建方法
        // $http.post("../specification/findAll").success(function (response) { //远程内置服务提交异步请求
        //     $scope.list = response;    //接受后台数据   json--> 数组      调用参数是遍历
        // })
       return $http.post("../specification/findAll");
    };

    //保存方法
    this.add=function (entity) {
        return $http.post("../specification/add",entity);
    };

    // this.addTbSpecificationOptions=function (entity) {
    //     return $http.post("../specification/add",entity);
    // }
    this.update=function (entity) {
        return $http.post("../specification/update",entity);
    };

    this.findOne = function (id) {
        return $http.get("../specification/findOne?id=" + id)
    };
    //删除
    this.dele = function (selectIds) {
        return $http.get("../specification/dele?Ids="+selectIds)
        };

    this.query = function (pageNo,pageSize,queryEntity) {
        return $http.post("../specification/query?pageNo="+pageNo+"&pageSize="+pageSize,queryEntity)
    };

});